### ISSUE TITLE
isearch (wrong-type-argument arrayp nil)

### ISSUE THREAD

**Issue by kuppo** [purcell/emacs.d#675]
I accounted this wired error whenever I hit `C-s` (key-binding for "isearch-forward")

Below is the backtrace:

> Debugger entered--Lisp error: (wrong-type-argument arrayp nil)
  window-font-width(nil nil)
  window-max-chars-per-line()
  isearch-update()
  isearch-mode(t nil nil nil)
  isearch-forward(nil 1)
  funcall-interactively(isearch-forward nil 1)
  call-interactively(isearch-forward nil nil)
  command-execute(isearch-forward)

Don't know what was going on there.

**Comment by kuppo** [comment]
Sorry, below is emacs and system version information:
> GNU Emacs 27.0.50 (build 1, x86_64-w64-mingw32) of 2018-06-20

> Windows 7 Enterprise Service Pack 1 64bit

**Comment by josephyoung** [comment]
I have the same issue. My OS is Windows 10, Chinese simplified.
Backtrace:
`Debugger entered--Lisp error: (wrong-type-argument arrayp nil)
  window-font-width(nil nil)
  window-max-chars-per-line()
  isearch-update()
  isearch-mode(nil nil nil nil)
  isearch-backward(nil 1)
  funcall-interactively(isearch-backward nil 1)
  call-interactively(isearch-backward nil nil)
  command-execute(isearch-backward)
`
Emacs version:

> GNU Emacs 26.1 (build 1, x86_64-w64-mingw32) of 2018-05-30

OS version:

> Windows 10 Home Edition, China
> Microsoft Windows [ver 10.0.17134.472]

> \HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\BuildLabEx = 17134.1.amd64fre.rs4_release.180410-1804

**Comment by josephyoung** [comment]
I trid to comment out the `(require 'init-locales)` in init.el, the isearch issue solved, but without the init-locales.el would cause a lot of other issues, please help.

**Comment by kuppo** [comment]
I may know what the problem is as `josephyoung` just remained me.

I mainly accounted the issue when I was using org-mode. In org-mode I inserted some `clock`s which are just timestamps corresponding to my encoding system. I am mainly using English on my OS, but the format settings are using `Chinese(Simplified, China)`, so the timestamp, I think, is not in UTF-8 encoding because every time I tried to save the org file, emacs flagged me some codes (mainly the day of the week) are not recognizable and asks me for saving encoding.

As `josephyoung` disabled init-locales then the issue not came, maybe we can do some changes to auto convert encoding to UTF-8 when saving files, what do you think, @purcell?

**Comment by kuppo** [comment]
Finally I think I fixed the issue, but I really don't know why, just post here for your reference.

I lied a bit about `I did alter the init.el file a bit, but I only changed some packages requirement.`. Beyond that, I also change my default font to [`Sarasa Mono SL`](https://github.com/be5invis/Sarasa-Gothic) that I download for displaying Chinese correctly.

From the traceback message, it said something to do with `window-font-width(nil nil)` that I just alter font for Emacs, it remains me. I disable my customized font, then the error disappeared.

But I don't want to disable the font, and I remember I used a package called [`cnfonts`](https://github.com/tumashu/cnfonts). In its TIPs, I found [this link](https://github.com/tumashu/cnfonts/issues/54#issuecomment-246228904) that says font setting also related to coding settings that `josephyoung` metioned the error also [related to coding system](https://github.com/purcell/emacs.d/issues/675#issuecomment-449931144). So I copied its setting and replace `init-locales` settings, restarted Emacs and isearch worked well, no more errors.

Hope it is useful for other people face the same issue. @josephyoung  @purcell
